### ISSUE TITLE
%N specifier doesn't parse millis correctly

### ISSUE THREAD

**Issue by gwenn** [time-rs/time#329]
May have been already fixed by v0.3 but:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=5a17d45369880434c77d75e9433f3dae
```rust
use time::PrimitiveDateTime; // 0.2.26

fn main() {
    let input = "2013-10-07 04:23:19.120";
    let result = PrimitiveDateTime::parse(input, "%F %T.%N");
    println!("{}", result.unwrap()); // 2013-10-07 4:23:19.000000120
}
```

**Comment by jhpratt** [comment]
This should actually return an error. `%N` is for nanoseconds and is always nine digits — using it to parse milliseconds is incorrect. This definitely doesn't apply to v0.3, as formatting and parsing has been redone entirely. I'll take a look at this tomorrow to see if I can get a patch released.

**Comment by gwenn** [comment]
For your information, Rfc3339 works fine:
```rust
use time::{Format, OffsetDateTime}; // 0.2.26

fn main() {
    let input = "2013-10-07T04:23:19.120-04:00";
    let result = OffsetDateTime::parse(input, Format::Rfc3339);
    println!("{}", result.unwrap()); // 2013-10-07 4:23:19.120 -4
}
```

**Comment by jhpratt** [comment]
And that behavior is correct. RFC3339 has its own parser, as it cannot be trivially represented by any other format description. That format parses any positive number of digits; `%N` _explicitly_ says it is always nine.

**Comment by gwenn** [comment]
You are right.

**Comment by jhpratt** [comment]
After quickly looking on mobile, I believe the error is [this line](https://github.com/time-rs/time/blob/6bf1f0b00f2d1fc56cbc580e63602afb3c6a68d4/src/format/parse.rs#L282). The 1 should be `num_digits - pad_size`. Which, now that I look at it is just `num_digits`, but that's not the point.

**Comment by jhpratt** [comment]
Turns out that wasn't it. I just had to switch `Padding::None` to `Padding::Some` for the nanosecond parser.

Released as v0.2.27.
